WordPress/src/main/java/org/wordpress/android/ui/reader/subfilter/SubFilterViewModel.kt

```diff
@@ -0,0 +1,338 @@
+                    }
+                }
+
+                for (blog in followedBlogs) {
+                    filterList.add(Site(
+                            onClickAction = ::onSubfilterClicked,
+                            blog = blog,
+                            isSelected = false
+                    ))
+                }
+            }
+
+            val tags = ReaderTagTable.getFollowedTags()
+
+            for (tag in tags) {
+                filterList.add(Tag(
+                        onClickAction = ::onSubfilterClicked,
+                        tag = tag,
+                        isSelected = false
+                ))
+            }
+
+            withContext(mainDispatcher) {
+                _subFilters.value = filterList
+            }
+        }
+    }
+
+    private fun onSubfilterClicked(filter: SubfilterListItem) {
+        _changeBottomSheetVisibility.postValue(Event(false))
+        updateSubfilter(filter)
+    }
+
+    fun changeSubfiltersVisibility(show: Boolean) {
+        val isCurrentSubfilterTracked = getCurrentSubfilterValue().isTrackedItem
+        val isSubfilterListTrackerRunning = readerTracker.isRunning(ReaderTrackerType.SUBFILTERED_LIST)
+
+        if (show) {
+            if (isCurrentSubfilterTracked && !isSubfilterListTrackerRunning) {
+                AppLog.d(T.READER, "TRACK READER ReaderPostListFragment > START Count SUBFILTERED_LIST")
+                readerTracker.start(ReaderTrackerType.SUBFILTERED_LIST)
+            } else if (!isCurrentSubfilterTracked && isSubfilterListTrackerRunning) {
+                AppLog.d(T.READER, "TRACK READER ReaderPostListFragment > STOP Count SUBFILTERED_LIST")
+                readerTracker.stop(ReaderTrackerType.SUBFILTERED_LIST)
+            }
+        } else if (isSubfilterListTrackerRunning) {
+            AppLog.d(T.READER, "TRACK READER ReaderPostListFragment > STOP Count SUBFILTERED_LIST")
+            readerTracker.stop(ReaderTrackerType.SUBFILTERED_LIST)
+        }
+
+        _shouldShowSubFilters.postValue(show)
+    }
+
+    fun getCurrentSubfilterValue(): SubfilterListItem {
+        return _currentSubFilter.value ?: appPrefsWrapper.getReaderSubfilter().let {
+            subfilterListItemMapper.fromJson(
+                    json = it,
+                    onClickAction = ::onSubfilterClicked,
+                    isSelected = true
+            )
+        }
+    }
+
+    fun setSubfilterFromTag(tag: ReaderTag) {
+        updateSubfilter(
+                Tag(
+                        onClickAction = ::onSubfilterClicked,
+                        tag = tag,
+                        isSelected = true
+                ))
+    }
+
+    fun setDefaultSubfilter() {
+        updateSubfilter(
+                SiteAll(
+                        onClickAction = ::onSubfilterClicked,
+                        isSelected = true
+                ))
+    }
+
+    fun onSubFiltersListButtonClicked() {
+        _updateTagsAndSites.value = Event(EnumSet.of(
```

Comment:
Do we really need to fetch the tags from the server whenever the user clicks on the subfilter button? Wdyt?

Reply:
I think this was a sort of easy win to get the list updated when you follow a site and come back immediately to the filter bottom sheet (IIRC sometimes it could take a bit to get the list updated if not triggered explicitly), but yep I agree it can be too much so I think it's fine to change.

Reply:
ok, I think we can update that since I believe the event bus event is emitted when you start following a site. So the list should get updated automatically. However, let's keep this out of scope of this PR or even this refactoring.